### PR TITLE
Convert image alt long rule to js

### PR DESCRIPTION
**Pull request by pattonwebz**
This pull request includes significant changes to the image alt text length validation functionality. The primary changes involve migrating the validation logic from PHP to JavaScript and updating the relevant rules and tests accordingly.

Migration from PHP to JavaScript:

* [`includes/rules.php`](diffhunk://#diff-40652b9b45019a70bc2344f1b523a3220a53ef38d2d326f6821588666a1d2e7aR227): Added a new `ruleset` attribute to the `img_alt_long` rule configuration.
* [`includes/rules/img_alt_long.php`](diffhunk://#diff-affb8cb0c90ae4959955297fb8f4e0472485ca38f0210a454f7063a4d2e4d36fL1-L40): Removed the PHP function `edac_rule_img_alt_long` that handled the image alt text length validation.
* [`src/pageScanner/checks/img-alt-long-check.js`](diffhunk://#diff-6e6317ec29c2839efbddf39730cb1c629688ad48ca273d7bd98726a5a302d253R1-R34): Added a new JavaScript check for validating image alt text length, porting the logic from the removed PHP function.
* [`src/pageScanner/rules/img-alt-long.js`](diffhunk://#diff-a322a1218a30cb9b3213b2bbe17bec13274ca0fb437f2c777e560c21eabb46e3R1-R18): Added a new rule for detecting images with excessively long alt text, based on WCAG 1.1.1.

Integration and Testing:

* [`src/pageScanner/index.js`](diffhunk://#diff-dfb86049e76745f75b4b756a00137c09d0ea1b1bb7bd8d3db94e8c5f38728c10R43-R44): Integrated the new `img_alt_long_check` and `imageAltLong` rule into the page scanner's configuration. [[1]](diffhunk://#diff-dfb86049e76745f75b4b756a00137c09d0ea1b1bb7bd8d3db94e8c5f38728c10R43-R44) [[2]](diffhunk://#diff-dfb86049e76745f75b4b756a00137c09d0ea1b1bb7bd8d3db94e8c5f38728c10R97) [[3]](diffhunk://#diff-dfb86049e76745f75b4b756a00137c09d0ea1b1bb7bd8d3db94e8c5f38728c10R119) [[4]](diffhunk://#diff-dfb86049e76745f75b4b756a00137c09d0ea1b1bb7bd8d3db94e8c5f38728c10R157)
* [`tests/jest/rules/imgAltLong.test.js`](diffhunk://#diff-46303f8bb7944f00c4243f1453e733feee7536f00ac0c50b856b6f21d1e82836R1-R113): Added comprehensive tests for the new image alt text length validation rule, covering various scenarios and edge cases.

## Checklist

- [x] PR is linked to the main issue in the repo
- [x] Tests are added that cover changes

Closes: #905

### CODE REVIEW

File: tests/jest/rules/imgAltLong.test.js [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+			html: `<img src="test.jpg" alt="${ 'b'.repeat( 100 ) }">`,
+			window: {
+				scanOptions: {
+					maxAltLength: 50,
+				},
+			},
+			shouldPass: false,
+		},
+		{
+			name: `should pass when alt is 400 characters and scanOptions.maxAltLength is set to 50`,
```
Comment (SteveJonesDev) [inline tests/jest/rules/imgAltLong.test.js:86]:
```suggestion
			name: `should pass when alt is 400 characters and scanOptions.maxAltLength is set to 500`,
```

**Review by SteveJonesDev — CHANGES_REQUESTED** [review]
I didn't find this rule on the a11ycheckrules website. I ran the tests through a WordPress post and it worked correct. I tested setting a custom `edac_max_alt_length` filter as well. Just one minor typo and this is good to go.
